service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java

```diff
@@ -198,44 +199,61 @@ public String toString() {
-        LOGGER.debug("Registering data request: {}", request);
-
+        final String uuid = UUID.randomUUID().toString();
+        LOGGER.debug("Registering data request: {}", request, uuid);
+        auditRequestReceived(request, uuid);
         final GetUserRequest userRequest = new GetUserRequest().userId(request.getUserId());
+        userRequest.setOriginalRequestId(uuid);
         LOGGER.debug("Getting user from userService: {}", userRequest);
         final CompletableFuture<User> futureUser = userService.getUser(userRequest)
                 .thenApply(user -> {
                     LOGGER.debug("Got user: {}", user);
                     return user;
+                })
+                .exceptionally(ex -> {
+                    LOGGER.debug("Failed to get user: {}" + ex.getMessage());
+                    auditRequestReceivedException(request, uuid, ex);
+                    throw new RuntimeException(ex); //rethrow the exception
                 });
-
         final GetResourcesByIdRequest resourceRequest = new GetResourcesByIdRequest().resourceId(request.getResourceId());
+        resourceRequest.setOriginalRequestId(uuid);
         LOGGER.debug("Getting resources from resourceService: {}", resourceRequest);
         final CompletableFuture<Map<LeafResource, ConnectionDetail>> futureResources = resourceService.getResourcesById(resourceRequest)
                 .thenApply(resources -> {
                     LOGGER.debug("Got resources: {}", resources);
                     return resources;
+                })
+                .exceptionally(ex -> {
+                    LOGGER.debug("Failed to get resources: {}" + ex.getMessage());
+                    auditRequestReceivedException(request, uuid, ex);
+                    throw new RuntimeException(ex); //rethrow the exception
                 });
-
         final RequestId requestId = new RequestId().id(request.getUserId().getId() + "-" + UUID.randomUUID().toString());
-
         final DataRequestConfig config = new DataRequestConfig();
         config.setContext(request.getContext());
-
+        config.setOriginalRequestId(uuid);
         return CompletableFuture.allOf(futureUser, futureResources)
-                .thenApply(t -> getPolicy(request, futureUser, futureResources))
+                .thenApply(t -> getPolicy(request, futureUser, futureResources, uuid))
                 .thenApply(multiPolicy -> ensureRecordRulesAvailableFor(multiPolicy, futureResources.join().keySet()))
                 .thenAccept(multiPolicy -> {
-                    audit(request, futureUser.join(), multiPolicy, AuditType.DATA_REQUEST_RECEIVED);
-                    cache(request, futureUser.join(), requestId, multiPolicy, futureResources.join().size());
-                }).thenApply(t -> {
+                    auditProcessingStarted(request, futureUser.join(), multiPolicy, uuid);
+                    cache(request, futureUser.join(), requestId, multiPolicy, futureResources.join().size(), uuid);
```

Comment:
I believe at this point the requestId is doing the same job as the uuid is. Therefore it might be worth using the requestId instead of the uuid.